src/validationFunctions.js

```diff
@@ -1,58 +1,72 @@
-    function isInvalidString (value) {
-        return typeof value !== "string"                                  
-    }
+// SIMPLE JS FUNCTIONS: 
+function isInvalidString(value) {
+    return typeof value !== "string"                                  
+}
     
-    function isInvalidNum(value) {  
+function isInvalidNum(value) {  
         return typeof(value) !== "number";  
-    }
+}
     
-    // 400 - BAD REQUEST
-    function generateErr400(message) { 
-        const newError = new Error(message);
-        newError.status = 400; 
-        return newError; 
-    } 
+// 400 - BAD REQUEST
+function generateErr400(message) { 
+    const newError = new Error(message);
+    newError.status = 400; 
+    return newError; 
+} 
 
-    // 404 - NOT FOUND
-    function generateErr404(id) {
-        let message = `cat id '${id}' not found in database`
-        const newError = new Error(message);
-        newError.status = 404; 
-        return newError; 
-    }
+// 404 - NOT FOUND
+function generateErr404(id) {
+    let message = `id '${id}' not found in database`
+    const newError = new Error(message);
+    newError.status = 404; 
+    return newError; 
+}
 
-    // checks for a populated object 
-    function checkObjFormat(keys) {
-        let message = (keys.length !== 0) ? 
-            'Check successful - request data verified as valid object' : 'Error: Request data is not a valid object. ' 
-        return message; 
-    } 
+// OBJECT CHECKS: to follow pattern -> return error OR success message string. Error message to be prepended with 'Error'.  
 
-    // checks for valid keys 
-    function checkObjKeys(keys) {
-        const validKeys = ["name", "ageInYears", "favouriteToy", "description", "breedId"];
-        let invalidKeys = keys.filter(key => { 
-            return !validKeys.includes(key)
-        })
-        let message = (invalidKeys.length === 0) ? 'Check successful - object keys valid' : `Error: Property '${invalidKeys[0]}' is invalid` ;
-        return message;  
-    }
+// checks for a populated object 
+function checkObjFormat(keys) {
+    let message = (keys.length !== 0) ? 
+        'Check successful - request data verified as valid object' : 'Error: Request data is not a valid object. ' 
+    return message; 
+} 
 
-    // checks for valid property values  
-    function checkObjValues(keys, object) {  
-        let isError = [];
-        keys.forEach((key) => {
-            let value = object[key]; 
-            let invalidParam = (key === "ageInYears") ? isInvalidNum(value) : isInvalidString(value);
-            if (invalidParam === true) {  
-                isError.push(`Error: Invalid ${key} parameter: "${value}"`)
-            }  
-        })      
-        let message = isError[0] || 'Check successful - object property values valid';
-        return message; 
-    }
+// checks for valid keys 
+function checkObjKeys(keys) {
+    const validKeys = ["name", "ageInYears", "favouriteToy", "description", "breedId"];
```

Comment:
If possible I think it'd be preferable to extract `validKeys` into a parameter for this function. This has the advantage that it'd allow you to remove `checkBreedObjKeys`, and it also means that `validationFunctions.js` is better encapsulated and doesn't require specific knowledge about how cat/breed objects are validated.

Reply:
ooh good idea!